docs/source/isolation.rst

```diff
@@ -0,0 +1,188 @@
+different Mesos agents, and how these isolation mechanisms are implemented.
+
+Note: Knowing the details of these systems isn't a requirement of using PaaSTA;
+most service authors may never need to know the details of such things. In
+fact, one of PaaSTA's primary design goals is to *hide* the minutiae of
+schedulers and resource isolation. However, this may benefit administrators
+of PaaSTA (and, more generally, Mesos clusters), and the simply curious.
+
+Final note: The details herein may, nay, will contain (unintended) inaccuracies.
+If you notice such a thing, we'd be super grateful if you could open a pull
+request to correct the document!
+
+How Tasks are Scheduled on Hosts
+--------------------------------
+
+To first understand how these resources are used, one must understand how
+a task is run on a Mesos cluster.
+
+Mesos can run in two modes: Master and Agent. When a node is running Mesos in
+Master mode, it is responsible for communicating between agent processes and
+frameworks. A Framework is a program which wants to run tasks on the Mesos
+cluster.
+
+A master is responsible for presenting frameworks with resource offers.
+Resource offers are compute resource free for a framework to run a task. The
+details of that compute resource comes from the agent nodes, which regularly
+tell the Master agent the resources it has available for running tasks. Using
+the correct parlance, Mesos agents make 'offers' to the master.
+
+At Yelp, the resource that a given agent has available is determined by a
+couple of things: if the node also runs 'classic' services, then the resources
+given to the agent are decided by a script which takes a rough guess at how
+much resource is used *without* Mesos running, and subtracts it from the total
+available resource. That gives a *very* approximate guess at the resources
+available for the Mesos Agent to run tasks.
+
+If, however, the agent node is a dedicated PaaSTA box (defined by having the
+puppet role ``paasta``), then all the resources available on the box are given
+to the Agent to run tasks. This means that the cpus offered is set to the
+total number of processors on the host, and the memory offered is equal to the
+total memory available on the host.
+
+Once a master node receives offers from an agent, it forwards it to
+a framework. Resource offers are split between frameworks according to
+the master's configuration - there may be particular priority given
+to some frameworks.
+
+At Yelp, we treat the frameworks we run (at the time of writing, Marathon and
+Chronos) equally. That means that frameworks *should* have offers distributed
+between them evenly, and all tasks are considered equal.
+
+It is then up to the framework to decide what it wants to do with an offer.
+The framework may decide to:
+
+  * Reject the offer, if the framework has no tasks to run.
+  * Reject the offer, if the resources included in the offer are not enough to
```

Comment:
I'd probably stress (here, on in the dedicated section) how the `cpu` setting (and similarly `mem`) in `marathon*.yaml` drives the reject/accept criteria.